web/controllers/pair_controller.ex

```diff
@@ -29,30 +29,37 @@ defmodule Pairmotron.PairController do
     |> render_index(year, week, groups_and_pairs)
   end
 
-  @spec render_index(Plug.Conn.t, integer(), 1..53, [{Types.group, Types.pair}]) :: Plug.Conn.t
+  @spec render_index(Plug.Conn.t, integer(), 1..53, [{Types.group, [{Types.pair, Types.retro | nil}]}]) :: Plug.Conn.t
   defp render_index(conn, year, week, groups_and_pairs) do
     render(conn, "index.html", year: year, week: week, groups_and_pairs: groups_and_pairs,
               start_date: Timex.from_iso_triplet({year, week, 1}),
               stop_date: Timex.from_iso_triplet({year, week, 7}))
   end
 
-  @spec fetch_groups_and_pairs(Types.user, integer(), 1..53) :: {[{Types.group, Types.pair}], [String.t]}
+  @spec fetch_groups_and_pairs(Types.user, integer(), 1..53) :: {[{Types.group, [{Types.pair, Types.retro | nil}]}], [String.t]}
   defp fetch_groups_and_pairs(user, year, week) do
     user = user |> Repo.preload(:groups)
-    {pairs, messages} = pairs_for_user_groups(user, year, week)
-    {Enum.zip(user.groups, pairs), messages}
+    groups_with_pairs_for_user(user, year, week)
   end
 
-  @spec pairs_for_user_groups(Types.user, integer(), 1..53) :: {[Types.pair], [String.t]}
-  defp pairs_for_user_groups(user, year, week) do
+  @spec groups_with_pairs_for_user(Types.user, integer(), 1..53) :: {[{Types.group, [{Types.pair, Types.retro | nil}]}], [String.t]}
```

Comment:
It's kind of sort of used in the GroupPairController. There's actually a bit of refactoring that should occur there as well to mirror the fact that technically in the PairController, it's now legal to be in multiple pairs in one week for one group. This results in some buggy behavior in the GroupPairController, which is fine for now since this state can only be reached if something has gone horribly wrong, or someone is poking the database manually.